lib/cylc/task_proxy.py

```diff
@@ -1508,9 +1508,13 @@ def get_state_summary(self):
         """Return a dict containing the state summary of this task proxy."""
         self.summary['state'] = self.state.status
         self.summary['spawned'] = str(self.has_spawned)
-        self.summary['mean_elapsed_time'] = (
-            float(sum(self.tdef.elapsed_times)) /
-            max(len(self.tdef.elapsed_times), 1))
+        count = len(self.tdef.elapsed_times)
+        if count == 0:
+            self.summary['mean_elapsed_time'] = \
+                max(self.summary['execution_time_limit'], 1)
+        else:
+            self.summary['mean_elapsed_time'] = (
+                float(sum(self.tdef.elapsed_times)) / count)
```

Comment:
I can have this default to 0 if that's preferable? I've used 1 here as a dummy value as it makes sense (to me) that if something runs its runtime is inherently a non-zero value. @hjoliver - any preference on a default before go change this?